src/constants.ts

```diff
@@ -51,3 +51,45 @@ export const CLASSIC_MODE_MYSTERY_ENCOUNTER_WAVES: [min: number, max: number] =
 
 /** Min - Max waves for mystery encounter in challenge mode. */
 export const CHALLENGE_MODE_MYSTERY_ENCOUNTER_WAVES: [min: number, max: number] = [10, 180];
+
+/**
+ * Spawn chance: (BASE_MYSTERY_ENCOUNTER_SPAWN_WEIGHT + WIGHT_INCREMENT_ON_SPAWN_MISS * <number of missed spawns>) / MYSTERY_ENCOUNTER_SPAWN_MAX_WEIGHT
+ */
+export const MYSTERY_ENCOUNTER_BASE_SPAWN_WEIGHT = 3;
+
+/**
+ * The divisor for determining ME spawns, defines the "maximum" weight required for a spawn
+ * If spawn_weight === MYSTERY_ENCOUNTER_SPAWN_MAX_WEIGHT, 100% chance to spawn a ME
+ */
+export const MYSTERY_ENCOUNTER_SPAWN_MAX_WEIGHT = 256;
+
+/**
+ * When an ME spawn roll fails, WEIGHT_INCREMENT_ON_SPAWN_MISS is added to future rolls for ME spawn checks.
+ * These values are cleared whenever the next ME spawns, and spawn weight returns to BASE_MYSTERY_ENCOUNTER_SPAWN_WEIGHT
+ */
+export const MYSTERY_ENCOUNTER_WEIGHT_INCREMENT_ON_SPAWN_MISS = 3;
+
+/**
+ * Specifies the target average for total ME spawns in a single Classic run.
+ * Used by anti-variance mechanic to check whether a run is above or below the target on a given wave.
+ */
+export const MYSTERY_ENCOUNTER_AVERAGE_ENCOUNTERS_PER_RUN_TARGET = 12;
+
+/**
+ * Will increase/decrease the chance of spawning a ME based on the current run's total MEs encountered vs AVERAGE_ENCOUNTERS_PER_RUN_TARGET
+ * Example:
+ * AVERAGE_ENCOUNTERS_PER_RUN_TARGET = 17 (expects avg 1 ME every 10 floors)
+ * ANTI_VARIANCE_WEIGHT_MODIFIER = 15
```

Comment:
```suggestion
 * MYSTERY_ENCOUNTER_AVERAGE_ENCOUNTERS_PER_RUN_TARGET = 17 (expects avg 1 ME every 10 floors)
 * MYSTERY_ENCOUNTER_ANTI_VARIANCE_WEIGHT_MODIFIER = 15
```

Should this example (and the numbers below) be adapted to the actual value of the average ME per run target? (so, 12)

Reply:
out of scope tbh. Something to note as an issue maybe?

Reply:
Nevermind, I misunderstood it